xml/hwbp_registers.xml

```diff
@@ -395,6 +412,257 @@
+        This register is accessible as \RcsrTdataOne when \FcsrTdataOneType is 6.
+
+        This replaces mcontrol in newer implementations and serves to provide additional
+        functionality.
+
+        Address and data trigger implementation are heavily dependent on how
+        the processor core is implemented. To accommodate various
+        implementations, execute, load, and store address/data triggers may fire at
+        whatever point in time is most convenient for the implementation.
+        The debugger may request specific timings as described in \FcsrMcontrolSixTiming.
+        Table~\ref{tab:hwbp_timing6} suggests timings for the best user experience.
+
+        \begin{table}[H]
+        \centering
+        \caption{Suggested Trigger Timings}
+        \label{tab:hwbp_timing6}
+        \begin{tabular}{|r|l|}
+        \hline
+        Match Type                  &amp; Suggested Trigger Timing \\
+        \hline
+        Execute Address             &amp; Before \\
+        Execute Instruction         &amp; Before \\
+        Execute Address+Instruction &amp; Before \\
+        Load Address                &amp; Before \\
+        Load Data                   &amp; After \\
+        Load Address+Data           &amp; After \\
+        Store Address               &amp; Before \\
+        Store Data                  &amp; Before \\
+        Store Address+Data          &amp; Before \\
+        \hline
+        \end{tabular}
+        \end{table}
+
+        A chain of triggers that don't all have the same \FcsrMcontrolSixTiming
+        value will never fire. That means to implement the suggestions in
+        Table~\ref{tab:hwbp_timing6}, both timings should be supported on load
+        address triggers.
+
+        This trigger type may be limited to address comparisons (\FcsrMcontrolSixSelect is
+        always 0) only. If that is the case, then \RcsrTdataTwo must be able to
+        hold all valid virtual addresses but it need not be capable of holding
+        other values.
+
+        In implementations that support \FcsrMcontrolSixMatch mode 1 (NAPOT), not all
+        NAPOT ranges may be supported.  All NAPOT ranges between $2^{1}$ and $2^{maskmax6}$
+        are supported where maskmax6 $\geq$ 1.  The value of maskmax6 can be determined
+        by the debugger via the following sequence:
+        \begin{enumerate}
+        \item Set mcontrol6.match=1.
+        \item Read mcontrol6.match. If it is not 1 then NAPOT matching is not supported.
+        \item Write all ones to tdata2.
+        \item Read tdata2. The value of maskmax6 is the index of the most significant 0 bit plus 1.
+        \end{enumerate}
+
+        <field name="type" bits="XLEN-1:XLEN-4" access="R" reset="6" />
+        <field name="dmode" bits="XLEN-5" access="WARL" reset="0" />
+        <field name="0" bits="XLEN-6:23" access="R" reset="0" />
+        <field name="sizehi" bits="22:21" access="WARL" reset="0">
```

Comment:
Yes.  Since software is going to have to support separate mcontrol and mcontrol6 formats anyway, it doesn't make sense to have this extra complication.  And I think everyone (hardware, software, spec readers) would be happier to have a unified field.